src/WordPress/DataStore.php

```diff
@@ -65,6 +66,27 @@ public function getHostAPIUserKey()
     }
 
     /**
+     * @return mixed
+     */
+    public function getDomainNameCache()
```

Comment:
You're duplicating some of your cache logic.  Why not do it all in WordPressAPI::getDomainList()? 
```
//CF\WordPress\WordPressAPI.php
    public function getDomainList($userId = null)
    {
        $cachedDomainName = $this->get(DataStore::CACHED_DOMAIN_NAME);
        if (empty($cachedDomainName)) {
             $cachedDomainName = $this->get(self::CACHED_DOMAIN_NAME);
             $this->set(DataStore::CACHED_DOMAIN_NAME, $cachedDomainName);
        }
        return array($cachedDomainName);
    }
```